src/core/createInstance.js

```diff
@@ -43,48 +42,59 @@ export default (
     }
     const config = createConfig(options);
 
-    try {
-      componentRegistry = initializeComponents(config);
-    } catch (e) {
-      configurationFailed = true;
-      throw e;
-    }
+    return initializeComponents(config);
   };
 
   const executeCommand = (commandName, options) => {
-    let command;
+    let execute;
 
-    if (configurationFailed) {
-      // We've decided if configuration fails we'll return
-      // never-resolved promises for all subsequent commands rather than
-      // throwing an error for each of them.
-      command = () => new Promise(() => {});
-    } else if (commandName === "configure") {
-      if (componentRegistry) {
+    if (commandName === "configure") {
+      if (configurePromise) {
         throw new Error(
           "The library has already been configured and may only be configured once."
         );
       }
-      command = configureCommand;
-    } else if (commandName === "log") {
-      command = logCommand;
+
+      execute = () => {
+        configurePromise = configureCommand(options);
+        return configurePromise;
+      };
     } else {
-      if (!componentRegistry) {
+      if (!configurePromise) {
         throw new Error(
           `The library must be configured first. Please do so by calling ${namespace}("configure", {...}).`
         );
       }
 
-      command = componentRegistry.getCommand(commandName);
-
-      if (!isFunction(command)) {
-        throw new Error(`The ${commandName} command does not exist.`);
+      if (commandName === "log") {
+        execute = () => logCommand(options);
+      } else {
+        execute = () => {
+          return configurePromise.then(
+            componentRegistry => {
+              const command = componentRegistry.getCommand(commandName);
+              if (!isFunction(command)) {
+                throw new Error(`The ${commandName} command does not exist.`);
+              }
+              return command(options);
+            },
+            () => {
+              // If configuration failed, we prevent the configuration
+              // error from bubbling here because we don't want the
+              // configuration error to be reported in the console every
+              // time any command is executed. Only having it bubble
+              // once when the configure command runs is sufficient.
+              // Instead, for this command, we'll just return a promise
+              // that never gets resolved.
+              return new Promise(() => {});
```

Comment:
Should we log something here like, "An error during configure is preventing this command from executing"

Reply:
I think it's a good idea. @jfkhoury and I didn't want it to be too naggy for each command attempted to be executed after configuration fails, but I think a warning would be good, so I've added logging for that. I'm still not sure if this is the right approach. In some ways, I think it would be the most technically correct to reject the promise for each command. But it does get quite naggy. I'm curious how you feel about it.

Reply:
Yes, there is already an error when the configure command fails.  If we had log levels these decisions would be easier.  I could see us logging every command call at a debug or trace level.

Reply:
We have the ability to log at a debug level. There's a `console.debug` command whose messages will only show up if you have "verbose" checked in the developer tools console (although I think that behavior varies by browser).